### PR TITLE
fix(storage): fix video asset resources on iOS13

### PR DESCRIPTION
**Pull request by facuacostag**
### Summary

Prior to iOS 13, when someone edited a video with the built-in system editor, It was just editing the length of the video and saving it as a new clip. On iOS 13, videos can be edited in multiple ways and therefore the `PHAssetResource` is different. When retrieving it we receive an `Adjustments.plist` file that contains the edition made, and two more files, one the video and another high quality version of the video.

The current implementation is like:

```
for (PHAssetResource *resource in resources) {
        exportSession.outputFileType = resource.uniformTypeIdentifier;
        if (exportSession.outputFileType != nil)
          break;
      }
```

When doing this, the main problem is that the `Adjustments.plist` has an `outputFileType` of  `PHAssetResourceTypeAdjustmentData` which passes as valid since we are just checking for not null values and then it fails when uploading to Firebase Storage.

Im not a native iOS developer so I don't know if this is the most optimised way of getting the proper edited video but I have tried It on iOS 13 and it works as expected.
### Checklist

- [x] Supports `iOS`

### Test Plan

1. Record a video with iOS camera
2. Edit it by adding a filter or changing length
3. Fetch it with CameraRoll library
4. Try to upload it to Firebase Storage.
5. Without this PR you will receive `-[AVAssetExportSession setOutputFileType:] Invalid output file type` because it will take the `Adjustments.plist outputFileType`
6. Try to do the same with this PR applied.
7. Works.

### Release Plan

[IOS][BUGFIX] [FirebaseStorage] - Get proper asset when uploading to Storage

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

File: packages/storage/ios/RNFBStorage/RNFBStorageCommon.m [hunk 1]
```diff
@@ -182,9 +182,17 @@ + (void)downloadAsset:(PHAsset *)asset toURL:(NSURL *)url completion:(void (^)(
...
       NSArray<PHAssetResource *> *resources = [PHAssetResource assetResourcesForAsset:asset];
       for (PHAssetResource *resource in resources) {
-        exportSession.outputFileType = resource.uniformTypeIdentifier;
-        if (exportSession.outputFileType != nil)
-          break;
+        if (resources.count > 1) {
+          if (resource.type != PHAssetResourceTypeVideo) {
+            continue;
+          } else {
+              exportSession.outputFileType = resource.uniformTypeIdentifier;
```
Comment (Salakar) [inline packages/storage/ios/RNFBStorage/RNFBStorageCommon.m]:
```suggestion
            exportSession.outputFileType = resource.uniformTypeIdentifier;
```

Indentation

File: packages/storage/ios/RNFBStorage/RNFBStorageCommon.m [hunk 2]
```diff
@@ -182,9 +182,17 @@ + (void)downloadAsset:(PHAsset *)asset toURL:(NSURL *)url completion:(void (^)(
...
+          }
+        } else {
+            exportSession.outputFileType = resource.uniformTypeIdentifier;
```
Comment (Salakar) [inline packages/storage/ios/RNFBStorage/RNFBStorageCommon.m]:
```suggestion
          exportSession.outputFileType = resource.uniformTypeIdentifier;
```

Indentation

**Comment by mikehardy** [comment]
ci.ios.test failure, I just re-ran it to see if it would resolve

details

```

[09:41:01]	[Step 3/5]  imageLabelerProcessImage()
[09:41:01]	[Step 3/5]  ✓ should throw if image path is not a string
[09:41:01]	[Step 3/5]  ✓ should return a local label array
[09:41:01]	[Step 3/5]  cloudImageLabelerProcessImage()
[09:41:01]	[Step 3/5]  ✓ should throw if image path is not a string
[09:45:21]	[Step 3/5] 
[09:45:21]	[Step 3/5] ⚠️ A test failed:
[09:45:21]	[Step 3/5] ️ -> should return a cloud label array
[09:45:21]	[Step 3/5] ️ -> Retrying in 1 seconds ... (1)
[09:49:43]	[Step 3/5]  Retry #1 failed...
[09:49:43]	[Step 3/5] ️ -> Retrying in 2 seconds ... (2)
[09:52:41]	[Step 3/5] The build Current::ci.ios.test #refs/pull/2750/head-503 {buildId=7391} has been running for more than 15 minutes. Terminating...
[09:52:41]	[Step 3/5] Execution timeout
[09:52:41]	[Step 3/5] Dumping threads before termination:
[09:52:41]	[Step 3/5] PID: 87825, PPID: 3205, Command line: sh /Users/ciuser/Library/buildAgent/temp/agentTmp/custom_script4020479148481864190
[09:52:41]	[Step 3/5] Failed to locate jps tool. No thread dump information is available
[09:52:41]	[Step 3/5] 
[09:52:41]	[Step 3/5] PID: 87829, PPID: 87825, Command line: node node_modules/.bin/nyc ./node_modules/.bin/detox test --configuration ios.sim.debug --cleanup
[09:52:41]	[Step 3/5] Failed to locate jps tool. No thread dump information is available
[09:52:41]	[Step 3/5] 
[09:52:41]	[Step 3/5] PID: 87832, PPID: 87829, Command line: /usr/local/bin/node /Users/ciuser/.node-spawn-wrap-87829-a259847438d2/node ./node_modules/.bin/detox test --configuration ios.sim.debug --cleanup
[09:52:41]	[Step 3/5] Failed to locate jps tool. No thread dump information is available
```

**Comment by Salakar** [comment]
Added simulator logs to CI, so should be able to see why the app is crashing shortly.

**Comment by Salakar** [comment]
Thank you again for this, will merge and land in v6.1

**Pull request merged**